### PR TITLE
Add isValidLastLocation() to check 

### LINKED ISSUE #110: Don't save or use "bad" last locations
The last location can get cached at a bad location such as `/desktop/hey.html` which gets displayed if Electron is not available. However, next time app is launched it will stay there.

This likely can happen for other error or similar pages we don't want to always go back to.

### PR DESCRIPTION
**Pull request by mkaz**
Creates the `isValidLastLocation()` function to use when navigating to and attempting to store the last location. We do not want to cache an error page or start back in the middle of the sign up flow.

Fix #110

### CODE REVIEW

File: desktop/server/index.js [hunk 1]
```diff
@@ -75,6 +75,21 @@ function startApp( started_cb ) {
 	} );
 }
 
+function isValidLastLocation( loc ) {
+	const invalids = [
+		'/desktop/hey.html',	// Page shown when no Electron
```
Comment (mjangda) [inline desktop/server/index.js]: Should we just block all entries in `/desktop/`?

File: desktop/server/index.js [hunk 2]
```diff
@@ -75,6 +75,21 @@ function startApp( started_cb ) {
...
+		'/start'				// Don't attempt to resume the signup flow
+	];
+
+	for ( let s of invalids ) {
```
Comment (mjangda) [inline desktop/server/index.js]: Can probably use the `every` helper:

```
return invalids.every( function( value ) {
    return ! loc.startsWith( value );
} );
```

**Comment by mjangda** [comment]
Two suggestions, but :+1:

File: desktop/server/index.js (hunk 1 above)
Comment (mkaz) [inline desktop/server/index.js]: Yeh, doesn't look like any of it should be stored, since most is just menus.

File: desktop/server/index.js (hunk 2 above)
Comment (mkaz) [inline desktop/server/index.js]: A bit too tricky for my taste. ;-)
Nice and concise but hard to mentally parse what will get returned and why

Comment (mjangda) [inline desktop/server/index.js]: Fair enough :)

**Pull request merged**